Makefile

```diff
@@ -8,7 +8,7 @@ export TERRAFORM_VERSION := 1.5.5
 
 export TERRAFORM_PROVIDER_SOURCE ?= hashicorp/vault
 export TERRAFORM_PROVIDER_REPO ?= https://github.com/hashicorp/terraform-provider-vault
-export TERRAFORM_PROVIDER_VERSION ?= 3.25.0
+export TERRAFORM_PROVIDER_VERSION ?= 4.2.0
```

Comment:
@jaylevin Are there any breaking changes we should be aware of?

Reply:
There is one "blocking change" that will prevent the deployment of new read policies that aren't set at the path level. The Vault Provider maintainers put together an upgrade guide for v4.0.0 and it's basically just updating the policy permissions to be set at the path level for mount metadata.

From the Vault Provider TF repository release notes from v3.25.0 -> 4.0.0:

> Important: This major version release includes performance improvements for deployments that manage many Vault secret or auth engine mounts. This release requires read policies to be set at the path level for mount metadata. For example, instead of permissions at sys/auth you must set permissions at the sys/auth/:path level. Please refer to the details in the [Terraform Vault Provider 4.0.0 Upgrade Guide](https://registry.terraform.io/providers/hashicorp/vault/latest/docs/guides/version_4_upgrade).

From the version 4.0.0 upgrade guide:

> The changes in this case are blocking but not destructive. That is, deployments will fail until the required Vault policy adjustments have been made.

Reply:
sounds good, release notes should take care of explaining that. @haarchri right?